public/javascripts/songButton.js

```diff
@@ -0,0 +1,146 @@
+var addOrDeleteSong = function(id) {
+    m.request({
+        method: 'POST',
+        url: '/songlist',
+        data: {
+            id: id
+        }
+    }).then(function(res) {
+        if (res.url) {
+            window.location.href = res.url
+        } else {
+            inLibrary(res.inLibrary)
+        }
+
+    })
+}
+
+var inLibrary = m.prop()
+var songButton = {
+    view: function() {
+        return m('button.btn', {
+            className: function() {
+                if (_.includes(inLibrary(), song._id)) {
```

Comment:
pull this out into a function and call it here. I noticed you use similar logic below, so reuse that function instead of copying and pasting.